### PR TITLE
CLN: standardize different freq message

### PR DESCRIPTION
**Pull request by jbrockmendel**
In #24282 some of the PeriodArray/PeriodIndex methods couldn't accept the decorator because of mismatches between _DIFFERENT_FREQ and DIFFERENT_FREQ_INDEX.  This unifies the two.

The remaining case to figure out is what to render for `other_freq` when dealing with an ndarray.  Right now the placeholder is "FIXME".

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Perhaps hold off on this for #24024, which is changing (some) of these to use `_check_compatiable_with`.

I suppose that would be a relatively easy fix to split off from #24024, and it'd mean fewer changes here as well.

File: pandas/core/arrays/period.py
Comment (TomAugspurger) [inline pandas/core/arrays/period.py]: Does "None" make sense here?

Comment (jbrockmendel) [reply pandas/core/arrays/period.py]: Either that or something resembling a least common denominator.  None sounds easier.

**Comment by jbrockmendel** [comment]
> I suppose that would be a relatively easy fix to split off from #24024, and it'd mean fewer changes here as well.

+1

File: pandas/_libs/tslibs/period.pyx
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx]: why don't you just make this a function and have it raise ``IncompatileFrequencey`` as well?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/period.pyx
Comment (jbrockmendel) [reply pandas/_libs/tslibs/period.pyx]: Sure.  I'm going to wait on this until after #24285 is merged.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @TomAugspurger if ok for you to merge ping (or merge)

File: pandas/core/arrays/period.py
Comment (TomAugspurger) [inline pandas/core/arrays/period.py]: `PeriodArray.__setitem__` is being removed, so I was hoping to not change these.

**Pull request merged**
